File: PythonLibs/retroWaveSynth.py
# retroWaveSynth.py - Uses Keras
from tensorflow import keras
from tensorflow.keras import layers

def build_model():
    model = keras.Sequential([
        layers.Dense(32, activation='relu', input_shape=(8,)),
        layers.Dense(1, activation='sigmoid')
    ])
    model.compile(optimizer='adam', loss='binary_crossentropy')
    print('Keras model summary:')
    model.summary()

if __name__ == '__main__':
    build_model()

File: C++Libs/retroWaveSynth.cpp
// retroWaveSynth.cpp - Uses tiny-dnn
#include "tiny_dnn/tiny_dnn.h"
#include <iostream>

int main() {
    tiny_dnn::network<tiny_dnn::sequential> nn;
    nn << tiny_dnn::layers::fc(4, 3) << tiny_dnn::activation::relu();
    std::cout << "Network topology created with tiny-dnn" << std::endl;
    return 0;
}
